evm/src/invariant_fuzz.rs

```diff
@@ -0,0 +1,360 @@
+                }
+
+                // Before each test, we must reset to the initial state
+                executor.borrow_mut().db = clean_db.clone();
+
+                Ok(())
+            })
+            .err()
+            .map(|test_error| InvariantFuzzError {
+                test_error,
+                // return_reason: return_reason.into_inner().expect("Reason must be set"),
+                return_reason: "".into(),
+                revert_reason: revert_reason.into_inner().expect("Revert error string must be set"),
+                addr: invariant_address,
+                func: ethers::prelude::Bytes::default(),
+            });
+
+        Some(InvariantFuzzTestResult {
+            invariants: invariant_doesnt_hold.into_inner(),
+            cases: FuzzedCases::new(fuzz_cases.into_inner()),
+        })
+    }
+
+    pub fn select_senders(&self, invariant_address: Address, abi: &Abi) -> Vec<Address> {
+        let mut selected: Vec<Address> = vec![];
+        if let Some(func) = abi.functions().into_iter().find(|func| func.name == "targetSenders") {
+            if let Ok(call_result) = self.evm.call::<Vec<Address>, _, _>(
+                self.sender,
+                invariant_address,
+                func.clone(),
+                (),
+                U256::zero(),
+                Some(abi),
+            ) {
+                selected = call_result.result;
+            } else {
+                warn!("The function targetSenders was found but there was an error querying addresses.");
+            }
+        };
+        selected
+    }
+
+    pub fn select_contracts(
+        &self,
+        invariant_address: Address,
+        abi: &Abi,
+    ) -> BTreeMap<Address, (String, Abi)> {
+        let mut selected: Vec<Address> = vec![];
+        if let Some(func) = abi.functions().into_iter().find(|func| func.name == "targetContracts")
+        {
+            if let Ok(call_result) = self.evm.call::<Vec<Address>, _, _>(
+                self.sender,
+                invariant_address,
+                func.clone(),
+                (),
+                U256::zero(),
+                Some(abi),
+            ) {
+                selected = call_result.result;
+            } else {
+                warn!("The function targetContracts was found but there was an error querying addresses.");
+            }
+        };
+
+        self.contracts
+            .clone()
+            .into_iter()
+            .filter(|(addr, _)| {
+                *addr != invariant_address &&
+                    *addr !=
+                        Address::from_slice(
+                            &hex::decode("7109709ECfa91a80626fF3989D68f67F5b1DD12D").unwrap(),
+                        ) &&
+                    *addr !=
+                        Address::from_slice(
+                            &hex::decode("000000000000000000636F6e736F6c652e6c6f67").unwrap(),
+                        ) &&
+                    (selected.is_empty() || selected.contains(addr))
```

Comment:
nice. glad this is optional. I hate dapptool's ux there, but agree its necessary